sql_joins/sql_joins.md

```diff
@@ -0,0 +1,875 @@
+
+## Forming a JOIN 
+
+There are two basic pieces of information you need to know to write successful joins:
+
+**Type of Join**
+-----
+
+What **type of join** do you want to use?  Let's say you have some students with math grades and some students with language grades.  Some students are in only the math table, some are only in the language table, and some are in both.  What part of the overlapping data do you want?  Only the data on students with both kinds of grades?  Or some other, larger set of student data?  
+
+The type of join shows up in SQL in a `FROM` statement, and will look something like:
+
+`FROM [table_1_name] [optional keyword] JOIN [table_2_name]`.
+
+**Join Criteria**
+----
+
+What **join criteria** would you like your join evaluated against?  In other words, how are you linking rows from one table to rows from another table?  What constitutes a "match" of data from one table to data from another?  If you're joining data that holds student grades, for example, you probably want to only join data together where the student ID matches.
+
+The join criteria shows up in SQL in an `ON` or `USING` statement, and will look something like:
+
+`ON [table_1_name.field_name] = [table_2_name.field_name]` 
+
+OR
+
+`USING (field_name)` (this method is less frequently used)
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+It can be surprisingly tricky to figure out what makes data "match" or "go together" in your join criteria.  For example, go back to the previous page and see if you can decide what data should be considered "matching" between the "depression\_scale" table and the "subject\_address" table.  Is it as simple as just matching on the subject id ("subj_id")?  Why or why not?
+
+</div>
+
+
+**Combining Join Type and Join Criteria**
+----
+
+You'll combine join type and join criteria by using both the `FROM` component and the `ON` or `USING` component.  Your SQL query might include lines that look something like this: 
+
+```sql
+...
+FROM table_1 LEFT JOIN table_2
+ON table_1.pat_id = table_2.pat_id
+...
+```
+
+For the rest of this module:
+
+* Let's first talk about join types, 
+* Then we'll explain join criteria, and 
+* We'll finish up with fuller examples to help you understand how these two elements of a join work together to give you the results you care about.
+
+### Join Types 
+
+Let's consider the gradebook example we mentioned earlier.  You are assembling grade reports for students.  You have two tables, one called "math\_grades" and one called "language\_grades".  Some students appear in "math\_grades", some in "language\_grades", and some students have rows in both tables.  Depending on your purposes, you might want any one of several types of joins, each with its own SQL keyword combination.  Let's imagine a Venn diagram, as shown below:
+
+<lia-keep>
+
```

Comment:
Do you know how a screen reader will handle this infographic? I'm not sure how or if the text in it becomes alt text.